### ISSUE TITLE
Sanity checking of uploaded files [anvi-server]

### ISSUE THREAD

**Issue by meren** [meren/anvio#268]
We are doing nothing about it. I will take a stab at it, and ask @paczian's help if I run into issues.

**Comment by meren** [comment]
Hi @paczian,

I am mostly done. In the latest codebase, when the user uploads incompatible files, we get errors back (I simply generate a mock interactive object to make sure the class is happy with what the user has provided). For instance, when I run `./start_server.sh mock` under `tests/server` directory, and upload these two files (which are under `tests/sandbox/files_for_manual_interactive/` directory):

![image](https://cloud.githubusercontent.com/assets/197307/12374580/469cb680-bc65-11e5-9475-67ea0c2da675.png)

For these files the upload function returns this:

``` python
{'status': 'error', 'message': 'Some split names found in your tree are missing in your view data. Hard to know what cuased this, but essentially your tree and your view does not seem to be compatible. Here is a couple of splits that appear in the tree but not in the view data: centrist, cathetus.', 'data': None}
```

At this point the user should see this, and the project should not be stored.

Also there is this other little issue: You can upload a project by only providing a tree file :) However, the upload should not be completed without a proper project name, and a data file.

**Comment by paczian** [comment]
Implemented in #282
